### PR TITLE
Align mutate functions in Redo sample

### PR DESCRIPTION
**Pull request by arv**
Make them match the batch endpoint.

### CODE REVIEW

**Comment by arv** [comment]
TBR

File: sample/redo/lib/main.dart [hunk 1]
```diff
@@ -183,50 +183,32 @@ class _MyHomePageState extends State<MyHomePage> {
...
-      double order = args['order'];
       final todo = await _read(tx, id);
       if (todo == null) {
         print('Warning: Possible conflict - Specified Todo $id is not present.'
             ' Skipping reorder.');
         return;
       }
-      todo.order = order;
+      todo.text ??= args['text'] ?? todo.text;
+      todo.complete = args['complete'] ?? todo.complete;
```
Comment (aboodman) [inline sample/redo/lib/main.dart:210]: why are the operators different on the different lines?

File: sample/redo/lib/main.dart [hunk 2]
```diff
@@ -183,50 +183,32 @@ class _MyHomePageState extends State<MyHomePage> {
...
+      todo.order = args['order'] ?? todo.order;
```
Comment (aboodman) [inline sample/redo/lib/main.dart:211]: [taste] This is correct, but I think the read/write pair in the context of replay is a bit difficult to reason about. Consider adjusting `write` to only write the supplied fields.

**Comment by aboodman** [comment]
See https://news.ycombinator.com/item?id=23027988 for meaning of tags.

File: sample/redo/lib/main.dart (hunk 1 above)
Comment (arv) [reply sample/redo/lib/main.dart:210]: bug

File: sample/redo/lib/main.dart (hunk 2 above)
Comment (aboodman) [reply sample/redo/lib/main.dart:211]: On second thought I take this back. Transactions definitely make this approach correct, and it's simple.

Comment (arv) [reply sample/redo/lib/main.dart:211]: `_write` takes a `Todo`, Not a "diff". If I did what you are suggesting I would have to read the `Todo` again inside `_write`. 

The `write` you seem to be suggesting seems to be what the `updateTodo_` function is doing?

Comment (aboodman) [reply sample/redo/lib/main.dart:211]: You're write (sorry not sorry).

Comment (aboodman) [reply sample/redo/lib/main.dart:211]: (I was thinking that the individual keys of a todo were stored separately in the kv store, not as a single value. nvm.)

**Pull request merged**